File: JAVA/S4/L1/binarysearch.java
import java.util.*;
class search{
    int bsearch(List<Integer>arr,int n,int key)
    {
        int l=0;
        int r=n-1;
        while(l<=r)
        {
            int mid=l+(r-l)/2;
            if(arr.get(mid)==key)return mid;
            if(arr.get(mid)<key)
            {
                l=mid+1;
            }
            else r=mid-1;
        }
        return -1;
    }
}
public class binarysearch{
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        List<Integer>arr=new ArrayList<>();
        int n=sc.nextInt();
        for(int i=0;i<n;i++)
        {
            int num=sc.nextInt();
            arr.add(num);
        }
        int key=sc.nextInt();
        search s=new search();
        int r=s.bsearch(arr,n,key);
        if(r==-1)System.out.print("no");
        else System.out.print(r);
    }
}

File: C/S4/L1/binarysearch.c
#include<stdio.h>
int search(int arr[],int l,int r,int key)
{
    while(l<=r)
    {
        int mid=l+(r-l)/2;
        if(arr[mid]==key)return mid;
        if(arr[mid]>key)
        {
            r=mid-1;
        }
        else l=mid+1;
    }
    return -1;
}
int main()
{
    int n;
    scanf("%d",&n);
    int arr[n];
    for(int i=0;i<n;i++)
    {
        scanf("%d",&arr[i]);
    }
    int key;
    scanf("%d",&key);
    int r=search(arr,0,n-1,key);
    if(r==-1)printf("not found");
    else printf("%d",r);
    return 0;

}